src/view/items/partial/getPartialTemplate.js

```diff
@@ -14,6 +14,11 @@ export default function getPartialTemplate ( ractive, name, parentFragment ) {
 		// parse and register to this ractive instance
 		let parsed = parser.parseFor( partial, ractive );
 
+		// register extra partials on the ractive instance if they don't already exist
+		for ( let k in parsed.p ) {
+			if ( !( k in ractive.partials ) ) ractive.partials[ k ] = parsed.p[ k ];
+		}
+
```

Comment:
Sorry late to the party on feedback, extend/fill-gap functionality already exists [here](https://github.com/ractivejs/ractive/blob/dev/src/utils/object.js#L78-L102).

Though not sure we should be using `for` `in` unless we actually need to go deeper into the prototype. My understanding is that it (for/in) prevents optimization of the function.